### PR TITLE
Pass additional files as inputs to CoreCompile

### LINKED ISSUE #2983: Changes to additional files don't seem to be taken into account by msbuild
1. Rebuild BasicCodeAnalysis.csproj from command line with msbuild:
    1. `cd \src\Compilers\VisualBasic\Portable\`
    2. `msbuild /m /t:rebuild`, verify build succeeds.

2. Now delete a few lines from any of the PublicAPI.txt files in the folder.

3. Build vb compiler using msbuild: `msbuild /m`

**Expected:** Build fails due to analyzer errors

**Actual:** Build is not even executed and shows succeeded with original assembly. It seems that msbuild OR our Vbc msbuild task is not even invoking the VB compiler when only the additional files for the project have changed from previous build. 
If you explicitly invoke vbc/vbc2, then the build does fail as expected.

I am not sure if this an msbuild bug or a bug in our msbuild task, so filing it on compilers area path.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #2983

### CODE REVIEW

File: src/Compilers/Core/MSBuildTask/Shared/Microsoft.CSharp.Core.targets [hunk 1]
```diff
@@ -54,6 +55,7 @@
       <Prefer32Bit>false</Prefer32Bit>
     </PropertyGroup>
 
+    <!-- TODO: Remove this ItemGroup once it has been moved to "_GenerateCompileInputs" target in Microsoft.Common.CurrentVersion.targets -->
```
Comment (mavasani) [inline src/Compilers/Core/MSBuildTask/Shared/Microsoft.CSharp.Core.targets]: I'll send a separate VSO PR to move this ItemGroup to Microsoft.Common.CurrentVersion.targets (with an additional line to clear out AdditionalFileItemNames at the end). When that change flows back, we can remove this ItemGroup from C# and VB specific core targets.

**Comment by mavasani** [comment]
Tagging @dotnet/roslyn-compiler @dotnet/roslyn-analysis for review.

**Comment by AlekseyTs** [comment]
LGTM

**Comment by mavasani** [comment]
@dotnet-bot retest windows_eta_open_prtest please
// Previous failure: http://dotnet-ci.cloudapp.net/job/Private/job/dotnet_roslyn-internal/job/master/job/windows_eta_open_prtest/816/
// Retest reason: Flaky build failure https://github.com/dotnet/roslyn/issues/12222

File: src/Compilers/Core/MSBuildTask/Shared/Microsoft.VisualBasic.Core.targets [hunk 2]
```diff
@@ -43,6 +44,7 @@
       <Prefer32Bit>false</Prefer32Bit>
     </PropertyGroup>
 
+    <!-- TODO: Remove this ItemGroup once it has been moved to "_GenerateCompileInputs" target in Microsoft.Common.CurrentVersion.targets -->
```
Comment (jaredpar) [inline src/Compilers/Core/MSBuildTask/Shared/Microsoft.VisualBasic.Core.targets]: Where is the bug tracking this?  Shouldn't add TODO without a tracking bug.

File: src/Compilers/Core/MSBuildTask/Shared/Microsoft.CSharp.Core.targets (hunk 1 above)
Comment (mavasani) [inline src/Compilers/Core/MSBuildTask/Shared/Microsoft.CSharp.Core.targets]: Filed https://github.com/dotnet/roslyn/issues/12223 for the TODO.

**Comment by agocke** [comment]
LGTM, once reference to bug has been added in comment

**Comment by mavasani** [comment]
@dotnet-bot retest perf_correctness_prtest please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn/job/master/job/perf_correctness_prtest/232/
// Retest reason: VSSDK Build error

**Comment by mavasani** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn/job/master/job/windows_debug_unit64_prtest/800/
// Retest reason: Issue https://github.com/dotnet/roslyn/issues/12236

**Comment by mavasani** [comment]
@agocke  - Done.

**Pull request merged**